pkg/kor/namespaces.go

```diff
@@ -0,0 +1,221 @@
+package kor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	"github.com/yonahd/kor/pkg/filters"
+)
+
+type GenericResource struct {
+	NamespacedName types.NamespacedName
+	GVR            schema.GroupVersionResource
+}
+
+func getGVR(name string, splitGV []string) (*schema.GroupVersionResource, error) {
+	switch NumberOfGVPartsFound := len(splitGV); NumberOfGVPartsFound {
+	case 1:
+		return &schema.GroupVersionResource{
+			Version:  splitGV[0],
+			Resource: name,
+		}, nil
+	case 2:
+		return &schema.GroupVersionResource{
+			Group:    splitGV[0],
+			Version:  splitGV[1],
+			Resource: name,
+		}, nil
+	default:
+		return nil, fmt.Errorf("gv is wrong length slice: %d", NumberOfGVPartsFound)
+	}
+}
+
+func ignoreResourceType(resource string, ignoreResourceTypes []string) bool {
+	for _, ignoreType := range ignoreResourceTypes {
+		if resource == ignoreType {
+			return true
+		}
+	}
+	return false
+}
+
+func getNamespacedResources(
```

Comment:
Let's rename this function as it no returns whether the namespace is empty

Reply:
renamed function